resources/views/components/repeater-table.blade.php

```diff
@@ -55,25 +55,21 @@
                         @foreach ($headers as $key => $header)
                             <th
                                 @class([
-                                    'filament-table-repeater-header-column p-2 text-left bg-gray-200/50 dark:bg-gray-900/60',
-                                    'ltr:rounded-l-xl rtl:rounded-r-xl' => ! $hasContainers && $loop->first,
-                                    'ltr:rounded-tl-xl rtl:rounded-tr-xl' => $hasContainers && $loop->first,
-                                    'ltr:rounded-r-xl rtl:rounded-l-xl' => ! $hasContainers && $loop->last && ! $hasActions,
-                                    'ltr:rounded-tr-xl rtl:rounded-tl-xl' => $hasContainers && $loop->last && ! $hasActions,
+                                    'filament-table-repeater-header-column px-3 py-2 font-medium text-left text-gray-600 dark:text-gray-300 bg-gray-200/50 dark:bg-gray-900/60',
+                                    'ltr:rounded-tl-xl rtl:rounded-tr-xl' => $loop->first,
+                                    'ltr:rounded-tr-xl rtl:rounded-tl-xl' => $loop->last && ! $hasActions,
                                 ])
                                 @if ($columnWidths && isset($columnWidths[$key]))
                                     style="width: {{ $columnWidths[$key] }}"
+                                @elseif (! $hasContainers)
+                                    style="width: {{ (100 - (10 * intval($hasActions))) / count($headers) }}%"
```

Comment:
This is to allocate 10% of table width to action column, and distribute the remaining 90% for other columns. No idea to make empty action col width being consistent with its actual width.

Reply:
This shouldn't even be needed. The header width can still be defined by the width set from the user and the colspan on the empty td should handle if fine.

Reply:
Just thinking very narrow table header for action col isn't really beautiful, but I'll agree with you.